utils/cache.js

```diff
@@ -0,0 +1,21 @@
+function removeSuggestionFromCache(cache, suggestion) {
+    var clean_cache = {};
+    for (key in cache) {
+        clean_cache[key] = cache[key].filter(function (element) {
+            return JSON.stringify(element).replace(/<\/?b>/g, '') !== JSON.stringify(suggestion).replace(/<\/?b>/g, '');
```

Comment:
This `.replace(/<\/?b>/g, '')` call is used 6 times in this PR, move to a function and reuse it